UpComingMovies/app/src/main/java/com/example/douglas/upcomingmovies/MainActivity.java

```diff
@@ -0,0 +1,221 @@
+    private TextView mErrorMessageDisplay;
+    private MovieServiceAdapter mMovieServiceAdapter;
+    private int lastPageRequested=1;
+    private final int MAX_PAGES=10;
+    private List<Movie> fetchedMovies;
+
+    /**
+     * This method will create everything needed for start the application.
+     * @param savedInstanceState
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        /**
+         * Recycler view make the action of calling views easier and it is more efficient than
+         * call findViewById everytime. It makes the application more smooth maybe even more
+         * user-friendly.
+         */
+        mRecyclerView = (RecyclerView) findViewById(R.id.rv_movieservice);
+        mErrorMessageDisplay = (TextView) findViewById(R.id.tv_error_message_display);
+
+        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
+        mRecyclerView.setLayoutManager(layoutManager);
+
+        mRecyclerView.setHasFixedSize(true);
+
+        mMovieServiceAdapter = new MovieServiceAdapter();
+
+        /**
+         * The number max of pages was locked for avoid overflow of data. This could be
+         * changed for a future release. Therefore this code means that everytime the user
+         * reach(by scrolling) the last movie, the application will do one more request.
+         */
+        mMovieServiceAdapter.setOnBottomReachedListener(new MovieServiceAdapter.OnBottomReachedListener() {
+            @Override
+            public void onBottomReached(int position) {
+                if(lastPageRequested < MAX_PAGES) {
+                    lastPageRequested+=1;
+                    loadMovieData();
+                }
+            }
+        });
+
+        /**
+         * The adapter is the component able to create new items and populates them on the screen
+         * with data.
+         */
+        mRecyclerView.setAdapter(mMovieServiceAdapter);
+
+        loadMovieData();
+    }
+
+    /**
+     * This AsyncTask is almost a must have because the application need to do all the computations
+     * before the draw of the next frame, creating a AsysncTask is a modular aproach leaving the
+     * application free to be rendered while the content is being fetched.
+     */
+    public class UpComingMovieServiceTask extends AsyncTask<URL,Void,List<Movie>>{
+
+        /**
+         * This method is reponsible for call the http request and parse the first part of the
+         * api's response.
+         *
+         * @param urls
+         * @return fetchedMovies
+         */
+        @Override
+        protected List<Movie> doInBackground(URL... urls) {
+            URL searchUrl = urls[0];
+            String movieServiceResults = null;
+
+            try {
+                movieServiceResults = NetworkUtils.getResponseFromHttpUrl(searchUrl);

```

Comment:
It was nice that you were worried about loading the content from the API without interfering with the view rendering. But, you could also have implemented this using tools like Retrofit or RxProgramming.

Reply:
Retrofit seems to be awesome! It is indeed one thing less to worry about.